apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-create-edit/store/dot-template.store.ts

```diff
@@ -113,26 +113,44 @@ export class DotTemplateStore extends ComponentStore<DotTemplateState> {
         }
     );
 
-    readonly saveTemplate = this.effect((origin$: Observable<DotTemplateItem>) => {
+    readonly saveAndPublishTemplate = this.effect((origin$: Observable<DotTemplateItem>) => {
+        return origin$.pipe(
+            switchMap((template: DotTemplateItem) => {
+                this.dotGlobalMessageService.loading(this.dotMessageService.get('publishing'));
+
+                return this.dotTemplateService.updateAndPublish(this.cleanTemplateItem(template));
+            }),
+            tap((template: DotTemplate) => {
```

Comment:
Should we use this instead: https://ngrx.io/guide/component-store/effect#tapresponse